dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdFileUploadTask.kt

```diff
@@ -0,0 +1,314 @@
+        }
+    }
+
+    private fun applyDatadogCiConfig(datadogCiFile: File) {
+        try {
+            val config = JSONObject(datadogCiFile.readText())
+            applyApiKeyFromDatadogCiConfig(config)
+            applySiteFromDatadogCiConfig(config)
+        } catch (e: JSONException) {
+            DdAndroidGradlePlugin.LOGGER.error("Failed to parse Datadog CI config file.", e)
+        }
+    }
+
+    private fun applyApiKeyFromDatadogCiConfig(config: JSONObject) {
+        val apiKey = config.optString(DATADOG_CI_API_KEY_PROPERTY, null)
+        if (!apiKey.isNullOrEmpty()) {
+            if (this.apiKeySource == ApiKeySource.GRADLE_PROPERTY) {
+                DdAndroidGradlePlugin.LOGGER.info(
+                    "API key found in Datadog CI config file, but it will be ignored," +
+                        " because also an explicit one was provided as a gradle property."
+                )
+            } else {
+                DdAndroidGradlePlugin.LOGGER.info("API key found in Datadog CI config file, using it.")
+                this.apiKey = apiKey
+                this.apiKeySource = ApiKeySource.DATADOG_CI_CONFIG_FILE
+            }
+        }
+    }
+
+    private fun applySiteFromDatadogCiConfig(config: JSONObject) {
+        if (this.site.isNotEmpty()) {
+            DdAndroidGradlePlugin.LOGGER.info(
+                "Site property found in Datadog CI config file, but it will be ignored," +
+                    " because also an explicit one was provided in extension."
+            )
+            return
+        }
+        val siteAsDomain = config.optString(DATADOG_CI_SITE_PROPERTY, null)
+        if (!siteAsDomain.isNullOrEmpty()) {
+            val site = DatadogSite.fromDomain(siteAsDomain)
+            if (site == null) {
+                DdAndroidGradlePlugin.LOGGER.warn("Unknown Datadog domain provided: $siteAsDomain, ignoring it.")
+            } else {
+                DdAndroidGradlePlugin.LOGGER.info("Site property found in Datadog CI config file, using it.")
+                this.site = site.name
+            }
+        }
+    }
+
+    @Suppress("CheckInternal")
+    private fun validateConfiguration() {
+        check(apiKey.isNotBlank()) { API_KEY_MISSING_ERROR }
+
+        if (site.isBlank()) {
+            site = DatadogSite.US1.name
+        } else {
+            val validSiteIds = DatadogSite.validIds
+            check(site in validSiteIds) {
+                "You need to provide a valid site (one of ${validSiteIds.joinToString()})"
+            }
+        }
+    }
+
+    private fun generateRepositoryFile(repositories: List<RepositoryInfo>) {
+        val data = JSONArray()
+        repositories.forEach {
+            data.put(it.toJson())
+            DdAndroidGradlePlugin.LOGGER.info(
+                "Detected repository:\n${it.toJson().toString(
+                    INDENT
+                )}"
+            )
+        }
+
+        val jsonObject = JSONObject()
+        jsonObject.put("version", 1)
```

Comment:
why do we hardcode the version here ?

Reply:
It's the supported version of the JSON payload to intake, but it should likely be a constant in the companion object. Will fix.